repos/system_upgrade/common/actors/systemd/checksystemdbrokensymlinks/libraries/checksystemdbrokensymlinks.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from leapp import reporting
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import SystemdBrokenSymlinksSource
+
+FMT_LIST_SEPARATOR = '\n - '
+
+
+def _report_broken_symlinks(symlinks):
+    summary = (
+        'Leapp detected broken systemd symlinks on the system.'
+        ' The unit the symlink was pointing to was probably removed.'
+        ' The symlinks will not be handled during the upgrade.'
+        ' The found broken symlinks: {}'
+    )
+
+    hint = (
+        'You can use systemctl disable to remove these symlinks'
+    )
+
+    reporting.create_report([
+        reporting.Title(
+            'Broken systemd symlinks found on the system'
+        ),
+        reporting.Summary(summary.format(_printable_symlinks(symlinks))),
+        reporting.Remediation(hint=hint),
+        reporting.Severity(reporting.Severity.LOW),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+    ])
+
+
+def _report_enabled_services_broken_symlinks(symlinks):
+    summary = (
+        'Leapp detected broken systemd symlinks with same filenames as enabled systemd services.'
+        " The symlinks don't point to the unit's unit file."
+        ' The found broken symlinks: {}'
+    )
+
+    hint = (
+        "You can use systemctl reenable to remove and recreate the unit's symlinks or"
+        " systemctl disable to remove them."
+    )
+
+    reporting.create_report([
+        reporting.Title(
+            'Broken systemd symlinks with enabled services found on the system'
+        ),
+        reporting.Summary(summary.format(_printable_symlinks(symlinks))),
+        reporting.Remediation(hint=hint),
+        reporting.Severity(reporting.Severity.MEDIUM),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+    ])
+
+
+def _printable_symlinks(symlinks):
+    return FMT_LIST_SEPARATOR + FMT_LIST_SEPARATOR.join(sorted(symlinks))
```

Comment:
it's better to define functions before the use, so people can do the top-down reading, instead of searching where the function is defined, then go back, etc.